gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegionEntry.java

```diff
@@ -1433,12 +1437,17 @@ final void _setValue(RegionEntryContext context, @Unretained final Object val) {
     // release old SerializedDiskBuffer explicitly for eager cleanup
     final boolean isOffHeap = isOffHeap();
     Object rawOldVal = null;
+    Object keyBefore = getRawKey();
+    boolean newEntry = (getValueAsToken() == Token.REMOVED_PHASE1);
```

Comment:
rawOldVal already gets value field below for non-GFE-offheap case, so getValueAsToken is only required for isOffHeap case

Reply:
Don't think the Token check will work for all cases. In particular AbstractRegionMap.initRecoveredEntry during initial recovery may never create a RegionEntry with a token. A better call for this could be in AbstractRegionEntry constructor, or HashRegionEntryCreator.newEntry. A problem with former can be that it may be an entry that never gets put into the region map but with latter RegionContext is not available. Likewise decrement for map clear may not happen.

A better option could be to track all the calls to CachePerfStats.incEntryCount and add inc/dec calls there: ARE.makeTombstone, LocalRegion.incTombstoneCount and the callers of ARM.incEntryCount (HDFS ones could be excluded I suppose).